simconvoi.cc

```diff
@@ -2733,6 +2724,24 @@ void convoi_t::vorfahren()
 	INT_CHECK("simconvoi 711");
 }
 
+// a helper function for convoi_t::vorfahren()
+// check reserved_tiles and clear it when the route and reserved_tiles are different.
+void convoi_t::check_reserved_tiles_match_as_route()
```

Comment:
`check_xxx` という名前は一般的に、条件に合致しない場合は例外を投げる関数に使います。（SimutransのC++ではそもそも例外を用いないのでほぼ使うことはないと思います。）

```suggestion
void convoi_t::clear_reserved_tile_if_not_matching_route()
```